### PR TITLE
Add apiserver main hostname

### PR DESCRIPTION
**Pull request by ereslibre**
Fixes: bsc#1039437

### CODE REVIEW

File: pillar/kube-master.sls [hunk 1]
```diff
@@ -1,4 +1,5 @@
 api:
   server:
+    main_name:   null
```
Comment (kiall) [inline pillar/kube-master.sls]: For clarity, I'd probably call this "external_name" or "external_fqdn"

File: salt/kubernetes-master/init.sls [hunk 2]
```diff
@@ -23,6 +23,11 @@ include:
   {% do extra_names.append("DNS: " + extra_name) %}
 {% endfor %}
 
+# add the hostname reported by the user
+{% if salt['pillar.get']('api:server:main_name') %}
+  {% do extra_names.append("DNS: " + pillar['api']['server']['main_name']) %}
+{% endif %}
+
```
Comment (inercia) [inline salt/kubernetes-master/init.sls]: So this is just repeating what you did in the previous block?

```
{% for extra_name in pillar['api']['server']['extra_names'] %}
  {% do extra_names.append("DNS: " + extra_name) %}
{% endfor %}
{% if salt['pillar.get']('api:server:main_name') %}
  {% do extra_names.append("DNS: " + pillar['api']['server']['main_name']) %}
{% endif %}
```

Comment (ereslibre) [inline salt/kubernetes-master/init.sls]: Yes, adding the `main_name` that will be renamed to `external_fqdn`

**Comment by ereslibre** [comment]
Closing this in favour of using `extra_names`.

**Pull request merged**